pkg/apis/configuration/validation/validation.go

```diff
@@ -105,6 +115,82 @@ func validateUpstreamLBMethod(lBMethod string, fieldPath *field.Path, isPlus boo
 	return allErrs
 }
 
+func validateUpstreamHealthCheck(hc *v1alpha1.HealthCheck, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if hc == nil || !hc.Enable {
+		return allErrs
+	}
+
+	allErrs = append(allErrs, validatePath(hc.Path, fieldPath.Child("path"))...)
+	allErrs = append(allErrs, validateTime(hc.Interval, fieldPath.Child("interval"))...)
+	allErrs = append(allErrs, validateTime(hc.Jitter, fieldPath.Child("jitter"))...)
+	allErrs = append(allErrs, validatePositiveInt(hc.Fails, fieldPath.Child("fails"))...)
+	allErrs = append(allErrs, validatePositiveInt(hc.Passes, fieldPath.Child("passes"))...)
+	allErrs = append(allErrs, validateTime(hc.ConnectTimeout, fieldPath.Child("connect-timeout"))...)
+	allErrs = append(allErrs, validateTime(hc.ReadTimeout, fieldPath.Child("read-timeout"))...)
+	allErrs = append(allErrs, validateTime(hc.SendTimeout, fieldPath.Child("send-timeout"))...)
+	allErrs = append(allErrs, validateStatusMatch(hc.StatusMatch, fieldPath.Child("statusMatch"))...)
+
+	for i, header := range hc.Headers {
+		idxPath := fieldPath.Child("headers").Index(i)
+		allErrs = append(allErrs, validateHeader(header, idxPath)...)
+	}
+
+	if hc.Port > 0 {
+		for _, msg := range validation.IsValidPortNum(int(hc.Port)) {
+			allErrs = append(allErrs, field.Invalid(fieldPath.Child("port"), hc.Port, msg))
+		}
+	}
+
+	return allErrs
+}
+
+const statusMatchFmt = "(\\d*|\\s|\\-|\\!\\s)*"
```

Comment:
for status match the validation rules should be 
possible ! followed by valid (in the sense that NGINX config parser doesn't through any errors) status codes (100-999) or valid status code ranges

Reply:
I'll update. I think there must be a space ` ` after the ! `! 500` instead of `!500`  as well. 👍